rust/kernel/static_key.rs

```diff
@@ -0,0 +1,127 @@
+// SPDX-License-Identifier: GPL-2.0
+
+// Copyright (C) 2024 Google LLC.
+
+//! Logic for static keys.
+
+use crate::bindings::*;
+
+/// Doc todo
+#[macro_export]
+#[cfg(target_arch = "x86_64")]
+#[cfg(not(CONFIG_HAVE_RUST_ASM_GOTO))]
+macro_rules! static_key_false {
+    ($key:path, $keytyp:ty, $field:ident) => {{
+        let mut output = 1u32;
+
+        core::arch::asm!(
+            r#"
+            1: .byte 0x0f,0x1f,0x44,0x00,0x00
+
+            .pushsection __jump_table,  "aw"
+            .balign 8
+            .long 1b - .
+            .long 3f - .
+            .quad {0} + {1} - .
+            .popsection
+
+            2: mov {2:e}, 0
```

Comment:
you may want to use att_syntax ;-)

Reply:
What is that?

Reply:
`options(att_syntax)`, here is an example:

```rust
unsafe fn i32_xadd(v: *mut i32, mut i: i32) -> i32 {
    // SAFETY: Per function safety requirement, the address of `v` is valid for "xadd".
    unsafe {
        asm!(
            lock_instr!("xaddl {i:e}, ({v})"),
            i = inout(reg) i,
            v = in(reg) v,
            options(att_syntax, preserves_flags),
        );
    }

    i
}
```

in https://lore.kernel.org/rust-for-linux/20240322233838.868874-2-boqun.feng@gmail.com/

Reply:
Kernel developers will prefer that, yeah... :)

Sadly, specifying every single time that may be painful. Perhaps we could have our own `asm!` that expands to the actual one plus that option, if that is possible, or otherwise it would be nice to have a `rustc` flag to always use that by default.

Reply:
Added to our wishlist.

Reply:
A `asm!` wrapper should be fairly straightforward. The `asm` macro deliberately allows multiple options to simplify cases where people want to always add `options(att_syntax)`. See rust-lang/rust#73227.

Reply:
Thanks Gary, then we should probably do it soon if we are going to start adding more `asm!` blocks. I would still like to have a flag (or similar) nevertheless, to avoid extra macros and to match the C compilers.